### PR TITLE
Better catesian product example showing n*m effect

### PR DESCRIPTION
**Pull request by craigtaverner**
Small update to docs with more informative example, and removal of term 'cross product'.

### CODE REVIEW

File: community/cypher/docs/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala [hunk 1]
```diff
@@ -195,9 +195,9 @@ class QueryPlanTest extends DocumentingTestBase {
       title = "Cartesian Product",
       text =
         """Produces a cross product of the two inputs -- each row coming from the left child will be matched with all the rows from the right child operator.
-          |The following query will join all the people with all the locations and return the cartesian product of the nodes with those labels.
+          |The following query will join all the people with all the teams and return the cartesian product of the nodes with those labels.
```
Comment (boggle) [inline community/cypher/docs/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala]: is join good English wording for this for the uninitiated user?

Comment (craigtaverner) [inline community/cypher/docs/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala]: My vote is to re-write the entire line. I only changed locations->teams to get the cartesian product to look better. The previous 5*1=5 was not informative. I suggest I write new text now too.

Comment (craigtaverner) [inline community/cypher/docs/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala]: In thinking about better language for the 'cartesian product' docs I updated, I decided to try clarify the difference between the terms 'cross product' and 'cartesian product'. And I find the mathematically, the 'cartesian product' matches what we do correctly, while 'cross product' does not. A cross product of two 3D vectors results in another 3D vector, and not a 3x3 matrix like a cartesian product does. I think the use of 'cross product' in our language has leaked in as a result of the similar term 'cross join' as synonymous for 'cartesian join' in some SQL docs. In fact I read in microsoft docs that a 'cross join' is defined as a 'cartesian product'. Unless someone argues against this, I will start removing instances of 'cross product' terminology from the docs.

Comment (boggle) [inline community/cypher/docs/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala]: Good observation, let's start with rewriting this instance :)

Comment (nawroth) [inline community/cypher/docs/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala]: +1 on getting rid of "join".

Do we know any typical case where users end up with a Cartesian Product by mistake.
I'm wondering if we could write something helpful around that.

**Pull request merged**